docs/Tutorials/Permissioning/Upgrade-Permissioning-Contract.md

```diff
@@ -53,58 +48,126 @@ yarn run build
+
+    * `RETAIN_ADMIN_CONTRACT=true` to retain the current admin list
+    * `RETAIN_NODE_RULES_CONTRACT=true` to retain the current Node rules contract
+    * `RETAIN_ACCOUNT_RULES_CONTRACT=true` to retain the current Account rules contract
 
-1. If they exist, rename the following environment variables:
+1. Legacy: If they exist, rename the following environment variables:
     * `PANTHEON_NODE_PERM_ACCOUNT` to `BESU_NODE_PERM_ACCOUNT`
     * `PANTHEON_NODE_PERM_KEY` to `BESU_NODE_PERM_KEY`
     * `PANTHEON_NODE_PERM_ENDPOINT` to `BESU_NODE_PERM_ENDPOINT`.
 
-### 4. Deploy the contract
+2. If updating from v1 to v2, the Node Ingress contract must be re-deployed
+
+    !!! important
+
+        This step is only required if upgrading from v1 of the node permissioning contract
+        to v2 (because the interface changed, a new NodeIngress contract must be deployed).
+    If updating from v1 to v2, delete the following environment variable:
+    * `NODE_INGRESS_CONTRACT_ADDRESS`.
+
+### 4. Optional: Export current allowlists
+
+!!! note
 
-If using a `.env` file to configure [environment variables](#3-update-environment-variables), then
-copy the file to the `permissioning-smart-contracts` directory.
+    These steps enable you to export the current allowlists to assist in updating.
+    You can set these lists via the dapp if you prefer.
 
-In the `permissioning-smart-contracts` directory, deploy the Admin, Rules, and ingress
-contracts:
+1. Export the current allowlists by setting the following environment variables:
+
+    ```bash
+    RETAIN_ADMIN_CONTRACT=true
+    RETAIN_NODE_RULES_CONTRACT=true
+    RETAIN_ACCOUNT_RULES_CONTRACT=true
+    ```
+
+1. Log the current rules to console:
+
+    ```bash
+    truffle migrate
+    ```
+
+    The migration scripts will log the existing rules to the console, but no contracts will be deployed.
+
+1. Set initial values for updated deployment:
+
+    Using the values logged in the previous step, set the following environment variables as required:
+
+    ```bash
+    INITIAL_ADMIN_ACCOUNTS=<list-of-admins>
+    INITIAL_ALLOWLISTED_ACCOUNTS=<list-of-accounts>
+    INITIAL_ALLOWLISTED_NODES=<list-of-enode-urls>
+    ```
+
+1. Update environment variables for contracts that are to be deployed. For example:
+
+    ```bash
+    RETAIN_ADMIN_CONTRACT=true
+    RETAIN_NODE_RULES_CONTRACT=false
+    RETAIN_ACCOUNT_RULES_CONTRACT=false
+    ```  
+
+### 5. Deploy the contracts
+
+In the `permissioning-smart-contracts` directory, deploy the contracts:
 
 ```bash
-yarn truffle migrate --reset
+truffle migrate
 ```
 
 !!! important
 
-    Copy the `NodeIngress` and `AccountIngress` contract addresses which are displayed in the
-    output. These must be specified when restarting the nodes.
+    If updating from v1 to v2, copy the `NodeIngress` contract address which is displayed in the output. This must be specified when restarting the Besu nodes.
```

Comment:
> which is displayed in the output
is a bit underspecific. Where would I see that?

Reply:
added examples